license-header-check/license-header-check

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A license header check
+The tool checks if all files in the repo contain the license header.
+NOTE: this script is for github actions only, you should not use it anywhere else.
+"""
+import os
+import subprocess
+import sys
+import fnmatch
+from argparse import ArgumentParser
+
+from github import Github
+
+# Default configs
+LICENSE_PATTERN = ["NVIDIA CORPORATION", "Licensed under the Apache License"]
+
+def getDiffFiles(owner, repo_name, pull_number, token):
+    gh = Github(token)
+    repo = gh.get_repo(f'{owner}/{repo_name}')
+    pr = repo.get_pull(pull_number)
+    changed_files = pr.get_files()
+    files = [file.filename for file in changed_files]
+    # checkout source code
+    head_ref = pr.head.ref
+    subprocess.run(f"git clone {repo.clone_url} /code && cd /code && git checkout {head_ref}", shell=True, check=True)
+
+    return files
+
+def filterFiles(files, excludes):
+    excludes = excludes.split('\n')
```

Comment:
how should people pass the patterns? a file or a string? use '\n' to split is not safe IMHO, the format checking should be strict. otherwise, it could match unrelated ones.